src/Microsoft.IdentityModel.Tokens.Saml/Saml/SamlSecurityTokenHandler.cs

```diff
@@ -368,12 +367,33 @@ protected virtual SamlConditions CreateConditions(SecurityTokenDescriptor tokenD
             else if (SetDefaultTimesOnTokenCreation)
                 conditions.NotOnOrAfter = DateTime.UtcNow + TimeSpan.FromMinutes(TokenLifetimeInMinutes);
 
-            if (!string.IsNullOrEmpty(tokenDescriptor.Audience))
-                conditions.Conditions.Add(new SamlAudienceRestrictionCondition(new Uri(tokenDescriptor.Audience)));
+            var uriList = CreateUriList(tokenDescriptor);
+            if (!uriList.IsNullOrEmpty())
+                conditions.Conditions.Add(new SamlAudienceRestrictionCondition(uriList));
 
             return conditions;
         }
 
+        private static List<Uri> CreateUriList(SecurityTokenDescriptor tokenDescriptor)
+        {
+            var uriList = new List<Uri>();
+            if (!tokenDescriptor.Audiences.IsNullOrEmpty())
+            {
+                foreach (var audience in tokenDescriptor.Audiences)
+                {
+                    if (audience != null)
+                        uriList.Add(new Uri(audience));
+                }
+
+                if(!string.IsNullOrWhiteSpace(tokenDescriptor.Audience) && !tokenDescriptor.Audiences.Contains(tokenDescriptor.Audience))
```

Comment:
here and two lines before, lets use IsNullOrEmpty, that is consistent with the changes elsewhere.

Reply:
changed